### PR TITLE
v3.0.0: fix pfm and add ibc hooks

### PR DESCRIPTION
**Pull request by faddat**
## Description and Motivation

This is the full v3 upgrade for migaloo, including upgrade handlers.  It should not yet be merged, because we want to check over the upgrade hander before merge.  On merge, this will form a new testnet and mainnet.  Governance proposal slated to go up 9/16/2023

## Related Issues

* #233 

---
## Checklist:

- [x] I have read [Migaloo's contribution guidelines](https://github.com/White-Whale-Defi-Platform/migaloo-chain/blob/main/docs/CONTRIBUTING.md).
- [x] My pull request has a sound title and description (not something vague like `Update index.md`)
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation.
- [x] The code is formatted properly `golangci-lint run ./... --fix`.

### CODE REVIEW

File: app/app.go
Comment (faddat) [inline app/app.go:140]: this is just appparams

File: app/app.go
Comment (faddat) [inline app/app.go:168]: @hoangdv2429 - is there a reason we are calling these old upgrades?  I think they are no longer needed.

File: app/app.go
Comment (faddat) [inline app/app.go:1115]: remove

File: app/app.go
Comment (hoangdv2429) [reply app/app.go:168]: No, I think we can remove old upgrade calls

Comment (faddat) [reply app/app.go:168]: OK, please remove them :)

Comment (expertdicer) [reply app/app.go:168]: Should I make a PR to this PR ? cc: @faddat

**Comment by expertdicer** [comment]
I think after merging the #246, we are good to go. cc: @faddat @hoangdv2429

**Comment by faddat** [comment]
Agree.  In the future please work with other codeowners if I'm not available, but looks like great work guys!

<3

**Review by expertdicer — DISMISSED** [review]

File: app/app.go
Comment (faddat) [inline app/app.go:537]: did we set the contractkeeper successfully?

**Comment by faddat** [comment]
moment while I check on the contract keeper

**Comment by faddat** [comment]
https://github.com/terra-money/core/blob/a03a0657c7430d32e6329d86de78bb4aab9a9aa7/app/app.go#L609C1-L610C52

File: app/app.go
Comment (faddat) [reply app/app.go:537]: Now we have.

**Comment by faddat** [comment]
@expertdicer - sir, I think that in my own re-review, I found two issues:

* the on-chain version for the upgrade needed to be updated to v3.0.0
* We hadn't set the contract keeper

I have fixed both of these and re-requested your review.

Thanks!

<3 -Jacob

**Review by sontrinh16 — CHANGES_REQUESTED** [review]
should add this after intialize value for TransferKeepr in app.go line 564: `app.RouterKeeper.SetTransferKeeper(app.TransferKeeper)`

File: app/app.go
Comment (faddat) [inline app/app.go:638]: @sontrinh16 -- doesn't this do the same thing as what you just added in your PR?

File: app/app.go
Comment (faddat) [inline app/app.go:565]: I think we don't need this, because we setibcmodule

File: app/app.go
Comment (sontrinh16) [reply app/app.go:638]: hmmm no, i think this func is for setting the actual value of the `TransferKeeper` of the `PFMKeeper` ( `RouterKeeper` in this case ) since the TransferKeeper is intialize after the PFMKeeper so the TransferKeeper we pass in [line 524](https://github.com/White-Whale-Defi-Platform/migaloo-chain/blob/ed23688f5ee208ff4320330fd4761fe984527e65/app/app.go#L524) are empty so we need to call `SetTransferKeeper` to set the newly initialized value of `TranferKeeper`

**Review by hoangdv2429 — APPROVED** [review]

**Review by expertdicer — APPROVED** [review]

File: .gitignore
Comment (hoangdv2429) [inline .gitignore:4]: isn't this suppose to be */.DS_Store

**Review by hoangdv2429 — CHANGES_REQUESTED** [review]

**Review by sontrinh16 — APPROVED** [review]

**Review by hoangdv2429 — APPROVED** [review]

**Review by hoangdv2429 — APPROVED** [review]

**Review by 0xFable — APPROVED** [review]

**Pull request merged**
